### PR TITLE
[WIP] Implement GetFlyteKitMetrics endpoint

### PR DESCRIPTION
**Pull request by Yicheng-Lu-llll**
# TL;DR
[WIP] This is part of Flytekit Metrics Exploration

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: pkg/manager/impl/metrics_manager.go [hunk 1]
```diff
@@ -663,13 +668,39 @@ func (m *MetricsManager) GetExecutionMetrics(ctx context.Context,
...
+
+	nodeExecution, err := m.nodeExecutionManager.GetNodeExecution(ctx, admin.NodeExecutionGetRequest{
+		Id: request.Id,
+	},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	blob, err := m.urlData.Get(ctx, nodeExecution.Closure.SpanUri)
```
Comment (Yicheng-Lu-llll) [inline pkg/manager/impl/metrics_manager.go:682]: TODO: `m.urlData.Get `generates the signed URL. We may not need it.
